### ISSUE TITLE
Support COPY INTO ... FROM a Kafka topic

### ISSUE THREAD

**Issue by flaneur2020** [datafuselabs/databend#7499]
**Summary**

most organizations have some kind of kafka deployments in their data pipelines. it'd be convenient to allow COPY INTO from a Kafka topic.

currently our COPY INTO statement support load files s3/https/ftp. however, Kafka can also be treated as a storage like s3 bucket, it persists messages for a pretty long duration (~2 weeks), it have partitions so we can process the consumptions in parallel.

**Comment by zhang2014** [comment]
For Kafka, the message queue is always un-order and unbounded. Under what circumstances do we think that the copy command is finished?

**Comment by flaneur2020** [comment]
the messages in a kafka partition is in order and have a unique offset number, we can limit the messages batch size in each COPY statement, like:

```
COPY INTO github_events FROM KAFKA TOPIC="ghevents.topic" LIMIT=1024 [OFFSET=512] FORMAT="json"
```

**Comment by BohuTANG** [comment]
I think kafka here is a special storage engine<Like fuse engine>, we need a special catalog, the requirements same as https://github.com/datafuselabs/databend/issues/6395#issuecomment-1237545409

**Comment by sundy-li** [comment]
Kafka Engine is interesting, but it might be useless.

Though ClickHouse had these kind of engines (kafka/rabbitmq...) for years, it's rarely used.

**Comment by BohuTANG** [comment]
> Kafka Engine is interesting, but it might be useless.
> 
> Though ClickHouse had these kind of engines (kafka/rabbitmq...) for years, it's rarely used. I think users do not want to have the ETL programs inside the DBMS engine, it's hard to control (error handling, offset reset ..)

Right, we won't do that at the current time.
